internal/interceptor/server.go

```diff
@@ -287,27 +304,22 @@ func (s *Server) HandleRoot(w http.ResponseWriter, r *http.Request) {
 	pData.Environment = selectEnvironmentFromMapping(odsConfig.BranchToEnvironmentMapping, pData.GitRef)
 	pData.Version = odsConfig.Version
 
-	rendered, err := renderPipeline(odsConfig, pData, s.TaskKind, s.TaskSuffix)
-	if err != nil {
-		msg := "Could not render pipeline definition"
-		log.Println(requestID, fmt.Sprintf("%s: %s", msg, err))
-		http.Error(w, msg, 500)
-		return
-	}
+	tknPipeline := assemblePipeline(odsConfig, pData, s.TaskKind, s.TaskSuffix)
 
-	jsonBytes, err := yaml.YAMLToJSON(rendered)
+	_, err = s.TektonClient.GetPipeline(r.Context(), pData.Name, metav1.GetOptions{})
 	if err != nil {
-		msg := "could not convert YAML to JSON"
+		_, err := s.TektonClient.CreatePipeline(r.Context(), tknPipeline, metav1.CreateOptions{})
+		if err != nil {
+			msg := fmt.Sprintf("cannot create pipeline %s", tknPipeline.Name)
+			log.Println(requestID, fmt.Sprintf("%s: %s", msg, err))
+			http.Error(w, msg, http.StatusInternalServerError)
+			return
+		}
+	} else {
+		_, err := s.TektonClient.UpdatePipeline(r.Context(), tknPipeline, metav1.UpdateOptions{})
+		msg := fmt.Sprintf("cannot update pipeline %s", tknPipeline.Name)
```

Comment:
Should this really always fail or just if `err != nil?`, i.e. the pipeline cannot be updated?

Reply:
No 😃 

I added this right before pushing without running the tests again. The tests failed so at least it was caught. Anyway, fixed in 7029a40.